src/freertos_drivers/common/WiFiInterface.hxx

```diff
@@ -0,0 +1,377 @@
+        ACTIVE
+    };
+
+    /// WiFi Interface Operating mode.
+    enum class WiFiMode : uint8_t
+    {
+        /// Interface mode is not known.
+        UNKNOWN = 0,
+        /// Interface operates as a STATION (connects to access point).
+        STATION,
+        /// Interface operates as an ACCESS POINT (other STATIONS can connect
+        /// to it).
+        ACCESS_POINT,
+        /// Interface will operate as both a STATION and ACCESS POINT. If the
+        /// interface does not support this mode an error will be raised by the
+        /// @ref start method.
+        STATION_AND_ACCESS_POINT
+    };
+
+    /// Error codes used by the WiFiInterface APIs.
+    enum class WiFiErrorCode : uint8_t
+    {
+        /// API completed without error.
+        NO_ERROR = 0,
+        /// The requested API is not supported by the implementation.
+        UNSUPPORTED,
+        /// An invalid argument was passed to the API.
+        INVALID_ARGUMENT,
+
+        /// Requested SSID was not found via network scan or connection attempts.
+        SSID_NOT_FOUND,
+        /// Invalid password (or preshared key) was provided when connecting to
+        /// the requested access point.
+        INVALID_PASSWORD,
+        /// A timeout occurred for the SSID connection process.
+        CONNECT_TIMEOUT,
+
+        /// No networks were found via the network scan API.
+        NO_NETWORKS_FOUND,
+        /// The network scan operation timed out.
+        NETWORK_SCAN_TIMEOUT,
+        /// The network scan failed due to out of memory.
+        NETWORK_SCAN_NOMEM,
+        /// The network scan encountered a generic failure. This is WiFi
+        /// interface specific and should have corresponding log entries to
+        /// indicate the nature of the failure.
+        NETWORK_SCAN_GENERIC_FAILURE,
+        /// The network list is not available, this could indicate that no scan
+        /// was performed or the scan failed.
+        NETWORK_LIST_UNAVAILABLE,
+
+        /// The SSID profile was successfully stored.
+        PROFILE_CREATED,
+        /// The SSID profile was successfully removed.
+        PROFILE_DELETED,
+        /// A stored SSID profile was not found with the provided SSID or
+        /// the provided index is not valid.
+        PROFILE_NOT_FOUND,
+        /// The SSID profile could not be successfully stored as there was no
+        /// space.
+        PROFILE_OUT_OF_BOUNDS,
+        /// The SSID profile could not be successfully stored due to generic
+        /// failure. This is WiFi interface specific and should have
+        /// corresponding log entries to indicate the nature of the failure.
+        PROFILE_GENERIC_FAILURE,
+
+        /// The WiFi interface does not have an IPv4 address available.
+        IP_UNAVAILABLE,
+
+        /// RSSI value is not known or is otherwise unavailable.
+        RSSI_UNKNOWN
+    };
+
+    /// Contains the metadata for an access point.
+    struct NetworkListEntry
```

Comment:
Should we add bssid and channel members to this structure.  The CC32xx supports these (and more).  bssid, in case you don't know, is basically the MAC address of the ssid.